src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1897,8 +1897,42 @@ impl Cage {
         0 //chdir has succeeded!;
     }
 
-    //------------------------------------DUP & DUP2 SYSCALLS------------------------------------
-
+    ///##------------------------------------DUP & DUP2 SYSCALLS------------------------------------
+    /// ## `dup_syscall`
+    ///
+    /// ### Description
+    /// This function duplicates a file descriptor. It creates a new file
+    /// descriptor that refers to the same open file description as the original file descriptor.
+    /// * Finding the Next Available File Descriptor: If `start_desc` is provided and it is already in use, the function 
+    ///   will continue searching for the next available file descriptor starting from `start_desc`. If no file 
+    ///   descriptors are available, it will return an error (`ENFILE`).
+    /// * If `fd` is equal to `start_fd`, the function returns `start_fd` as the new file 
+    ///   descriptor. This is because in this scenario, the original and new file descriptors would point to the same 
+    ///   file description.
+    /// * The `_dup2_helper` function is called to perform the actual file descriptor duplication, handling the 
+    ///   allocation of a new file descriptor, updating the file descriptor table, and incrementing the reference count 
+    ///   of the file object.
+    /// * The function modifies the global `filedescriptortable` array, adding a new entry for the 
+    ///   duplicated file descriptor. It also increments the reference count of the file object associated with the 
+    ///   original file descriptor. 
+    /// * The `false` argument passed to `_dup2_helper` indicates that this call is from the `dup_syscall` function, 
+    ///   not the `dup2_syscall` function.
+    ///[dup(2)](https://man7.org/linux/man-pages/man2/dup.2.html)
```

Comment:
Just a minor change here, Would be great to add this manpage link after the errors section as it looks a bit mixed in the explanation here.